### PR TITLE
Updating Identity Engine > Embedded Authentication content for Node.js SDK Recover Password use case  

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Adding content for the [Recover Password](https://developer.okta.com/docs/guides/oie-embedded-sdk-use-cases/aspnet/oie-embedded-sdk-use-case-pwd-recovery-mfa/) use-case of the embedded authentication SDK use-case for the Node.js language.
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-408181](https://oktainc.atlassian.net/browse/OKTA-408181)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md [hunk 1]
```diff
@@ -1 +1,95 @@
...
+  const { authenticator } = req.body;
+  const authClient = getAuthClient(req);
+  const transaction = await authClient.idx[idxMethod]({ authenticator });
+  handleTransaction({ req, res, next, authClient, transaction });
+});
+```
+
+### Step 4: Create code verification page
+
+After the call to `idx.recoverPassword` has responded with a status of `Idx.Status:PENDING`, the user is redirected to a code verification page. The page must display a field to enter a code and a button or link to send the code to Okta for the email verification.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:59]:
```suggestion
After the call to `idx.recoverPassword()` has responded with a status of `Idx.Status:PENDING`, redirect the user to a code verification page. This page must display a field to enter a code and a button or link to send the code to Okta for the email verification.
```

Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:59]: `methodname()` format is normally used for calling out methods with parameters. But if this is not standard convention for node.js, please ignore this suggestion.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md [hunk 2]
```diff
@@ -1 +1,95 @@
...
+
+After the user checks their email for the code and enters the value into the field, they click a button to initiate the verification. Call `idx.recoverPassword` with the verification code passed in, as shown in the `authenticator.js` page of the SDK sample application.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:61]:
```suggestion
After the user checks their email for the code and enters the value into the field, they click a button to initiate the verification. Call the `idx.recoverPassword()` method with the verification code passed in, as shown in the `authenticator.js` page of the SDK sample application.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md [hunk 3]
```diff
@@ -1 +1,95 @@
...
+router.post('/challenge-authenticator/email', async (req, res, next) => {
+  const { idxMethod } = req.getFlowStates();
+  const { verificationCode } = req.body;
+  const authClient = getAuthClient(req);
+  const transaction = await authClient.idx[idxMethod]({ verificationCode });
+  handleTransaction({ req, res, next, authClient, transaction });
+});
+```
+
+If the `idx.recoverPassword` call is successful, it returns a status of `Idx.Status:PENDING` and the `nextStep` field indicates an input of a password value parameter. This status indicates that the user can now change their password. At this point, the user is redirected to the change password page.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:73]:
```suggestion
If the `idx.recoverPassword()` call is successful, it returns a status of `Idx.Status:PENDING` and the `nextStep` field indicates an input of a password value parameter. This status indicates that the user can now change their password. At this point, the user is redirected to a change password page.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md [hunk 4]
```diff
@@ -1 +1,95 @@
...
+
+### Step 5: Create change password page
+
+Create a change password page that allows the user to enter the new password and initiate the update.
+
+The final step is to make a call to `idx.recoverPassword`, passing in the value of the password, input by the user, as shown in the `recover-password.js` file of the SDK sample application. This call resets the user's password.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:79]:
```suggestion
The final step is to make a call to `idx.recoverPassword()`, passing in the value of the user input password, as shown in the `recover-password.js` file of the SDK sample application. This call resets the user's password.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md [hunk 5]
```diff
@@ -1 +1,95 @@
...
+    return;
+  }
+
+  const authClient = getAuthClient(req);
+  const transaction = await authClient.idx.recoverPassword({ password });
+  handleTransaction({ req, res, next, authClient, transaction });
+});
+```
+
+If the response to the `idx.recoverPassword` call returns `Idx.Status:SUCCESS` and tokens, the change password flow completed successfully and you can redirect the user to the default home page.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:95]:
```suggestion
If the response to the `idx.recoverPassword()` call returns `Idx.Status:SUCCESS` and tokens, the change password flow completed successfully and you can redirect the user to the default home page.
```

**Review by vanngo-okta — APPROVED** [review]
The use case is good. Most of my suggestions are formatting and some word choices. For the sequence diagram, try to make the arrow descriptions consistent (i.e. use command phrases).  I'm going to approve this PR. 🚢

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md [hunk 6]
```diff
@@ -1 +1,95 @@
...
\ No newline at end of file
+## Integration steps
+
+### Step 1: Add forgot password link to sign-in page
+
+The first step is to create a forgot password link on the sign-in page, as in the following example. This link points to the "Reset your password" page.
+
+<div class="common-image-format">
+
+![Forgot password screenshot where user can click link to reset password.](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-pwd-recovery-screenshot-forgot.png)
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:9]: Nice. Be good to review alt-text format as a team. Or at least I need to. Thanks.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md [hunk 7]
```diff
@@ -1 +1,95 @@
...
+```JavaScript
+router.post('/recover-password', async (req, res, next) => {
+  const { username } = req.body;
+  const authClient = getAuthClient(req);
+  const transaction = await authClient.idx.recoverPassword({ username });
+  handleTransaction({ req, res, next, authClient, transaction });
+});
+```
+
+The `handleTransaction` response, in the `handleTransaction.js` page, directs the user to a select authenticators page (`/select-authenticator` in this example) that the user needs to use to select an authentication type. In this case, the email factor is configured.
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:28]: I recast this bit to point to the correct page; the method is in another page and is referenced previously.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md (hunk 1 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:59]: I'm confirming. I've been simply stating to call it, and that's how I've done it all through my use-cases. I'll leave for now, but will clean this up on second pass.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md (hunk 2 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:61]: See next comment

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md [hunk 8]
```diff
@@ -1 +1,95 @@
...
+
+<div class="common-image-format">
+
+![Forgot password screenshot where user can click link to reset password.](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-pwd-recovery-screenshot-forgot.png)
+
+</div>
+
+Create a "Reset your password" page that initiates the reset password flow. This page accepts the user's email address and has a **Submit** button that starts the reset flow.
+
+After the user clicks **Submit** to start the reset flow, the next step is to call `idx.recoverPassword` and pass in the email address captured from the form. This starts the password recovery flow and returns a status of `Idx.Status:PENDING` and a `nextStep` field with inputs for an authenticator type.
```
[the remark this answers is not in the document]
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:15]: See comment later on

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md (hunk 5 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/nodejs/integrationsteps.md:95]: As per other comment on this terminology.

**Pull request merged**